matchms/SpectrumProcessor.py

```diff
@@ -0,0 +1,316 @@
+from collections import defaultdict
+from functools import partial
+from typing import Optional
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+import matchms.filtering as msfilters
+from matchms import Spectrum
+
+
+class SpectrumProcessor:
```

Comment:
Looks good to me! I will later try running the new pipeline on some larger test cases as well.